doc/modules/ROOT/pages/index.adoc

```diff
@@ -1,46 +1,235 @@
+----
+provider "azurerm" {
+  version = "=2.23.0"
+  features {}
+
+  subscription_id = "00000000-0000-0000-0000-000000000000"
+  tenant_id       = "11111111-1111-1111-1111-111111111111"
+
+}
+
+If you want to authenticate through service principals, please refer to https://www.terraform.io/docs/providers/azurerm/index.html#authenticating-to-azure[Authenticating to Azure].
+----
 
-////
-Write the steps in your guide here by making each step a different section
-////
+
+The account or service principal you use should have the role `Owner` assigned to it.
+
+--
+
+
+GCP::
++
+--
+
+You can use service accounts to authenticate Terraform. Get a service account key file, you can create key files on https://console.cloud.google.com/apis/credentials/serviceaccountkey[Google Console]. After you have created a key file, put its path  in `gcp/main.tf` as follows.
+
+[source,terraform]
+----
+provider "google" {
+  version = "3.5.0"
+
+  credentials = file("KEY-FILE-PATH/YOUR-KEY-FILE.json")
+  batching {
+    enable_batching = "false"
+  }
+  project = var.project_id
+  region  = var.region
+  zone    = var.zone
+}
+
+----
+The service account you use should have the following roles assigned to it:
+
+----
+Compute Admin
+Role Administrator
+Security Admin
+Service Account Admin
+Service Account User 
+----
+--
+====
+== Configuring Terraform for Connection
+
+Now Terraform has access to your credentials, we need to supply some variables to configure Terraform to be able to create resources correctly.
+
+Terraform will need a public-private key pair to be able to provision files and execute commands on the remote machines. For this you can use one of your existing key pairs or create a new one with the following command:
+
+[source, shell]
+----
+$ ssh-keygen -f ~/.ssh/YOUR-KEY-NAME -t rsa
+----
+
+This command will create two key files `YOUR-KEY-NAME.pub` and `YOUR-KEY-NAME`, public and private keys respectively. Terraform will use them to access VMs. 
+[tabs]
+====
+AWS::
++
+-- 
+
+Open `aws/terraform.tfvars` you will provide values for two variables.
+
+- `aws_key_name:` This is the name of the public-private key pair we created earlier.
+- `local_key_path:` This is the path we created the key pair at. 
+
+NOTE: In this guide we are using an ubuntu image, AWS by default creates a user with name `ubuntu`. So if you want to connect to your VMs by `ssh` you will have to use `ubuntu`. If you want to use another linux distribution, please refer to https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/managing-users.html[AWS EC2 Managing users] and change the variable `aws_ssh_user` accordingly.
+
+NOTE: The configuration defined in `aws/main.tf` assumes you have a default VPC in the region defined in `var.region`. For each region activated, AWS creates a default VPC, if you didn't delete it you can ignore this note. Otherwise please refer to https://docs.aws.amazon.com/vpc/latest/userguide/default-vpc.html#create-default-vpc[AWS Creating Default VPC].
```

Comment:
```suggestion
NOTE: The configuration defined in `aws/main.tf` assumes you have a default VPC in the region defined in `var.region`. AWS creates a default VPC for each activated region.  If you didn't delete it you can skip this note. Otherwise, please refer to https://docs.aws.amazon.com/vpc/latest/userguide/default-vpc.html#create-default-vpc[AWS Creating Default VPC].
```